src/Store/BuildStore.php

```diff
@@ -517,15 +517,30 @@ public function deleteAllByProject($projectId)
         return $q->rowCount();
     }
 
-    public function getOldByProject($projectId, $keep = 100)
+    /**
+     * Get old Builds by project.
+     *     - Get a batch of projects past the keep threshold.
+     *     - Reasonable batch size of 100.
+     *
+     * @param  int  $projectId
+     * @param  int  $keep
+     * @param  int  $batchSize
+     * @return array
+     */
+    public function getOldByProject($projectId, $keep = 100, $batchSize = 100)
     {
         if (is_null($projectId)) {
             throw new HttpException('Value passed to ' . __FUNCTION__ . ' cannot be null.');
         }
 
-        $query = 'SELECT * FROM {{' . $this->tableName . '}} WHERE {{project_id}} = :project_id ORDER BY {{create_date}} DESC OFFSET :keep';
+        $query = 'SELECT * FROM {{' . $this->tableName . '}} ' .
```

Comment:
You should delete the batch of builds from the end of the set. Otherwise, you will have gaps.